java/vector/src/main/java/org/apache/arrow/vector/ipc/JsonFileWriter.java

```diff
@@ -173,9 +176,9 @@ private void writeFromVectorIntoJson(Field field, FieldVector vector) throws IOE
         for (int i = 0; i < bufferValueCount; i++) {
           if (bufferType.equals(DATA) && (vector.getMinorType() == Types.MinorType.VARCHAR ||
                   vector.getMinorType() == Types.MinorType.VARBINARY)) {
-            writeValueToGenerator(bufferType, vectorBuffer, vectorBuffers.get(v-1), vector, i, scale);
+            writeValueToGenerator(bufferType, vectorBuffer, vectorBuffers.get(v-1), vector, i, scale, byteWidth);
```

Comment:
This method doesn't need to take `scale` and `byteWidth`. It can figure this out from `vector`.

Reply:
good catch, modified.